File: P1 2023.1/Questao(2)/python/Q2_2023_1.py
class Pessoa:
    def __init__(self, nome: str, peso: float, altura: float):
        self.nome = nome
        self.peso = peso
        self.altura = altura


def main():
    lista_pessoas = list()
    for i in range(0, 1000):
        string = input("Insira nome#peso#altura e exit para sair: ")

        if string != "exit":
            dados_pessoa = string.split("#")
            lista_pessoas.append(Pessoa(dados_pessoa[0],
                                        float(dados_pessoa[1]),
                                        float(dados_pessoa[2])))
        else:
            break

    total_pesos = float()

    for p in lista_pessoas:
        total_pesos += p.peso

    print("Media Pesos:", total_pesos / len(lista_pessoas))


if __name__ == "__main__":
    main()

File: P1 2023.1/Questao(2)/java/Q2_2023_1.java
import java.util.Scanner;

public class Q2_2023_1 {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int maxNumber = 1000, contador = 0;
        double somaPeso = 0;
        String[][] matrizPessoas = new String[maxNumber][3];

        System.out.println("Formato de entrada: nome#peso#altura");
        System.out.println("Para sair inserir: \"finalizar\"\n");

        String line = sc.nextLine();
        while (!line.equalsIgnoreCase("finalizar") && contador != maxNumber) {
            matrizPessoas[contador] = line.split("#");
            line = sc.nextLine();
            contador++;
        }

        for (int i = 0; i < contador; i++) {
            somaPeso += Double.parseDouble(matrizPessoas[i][1]);
        }
        System.out.println("Soma: " + somaPeso);
        System.out.println("Media:" + (somaPeso / contador));
        sc.close();
    }
}
